main.go

```diff
@@ -50,19 +55,40 @@ func main() {
 	}
 
 	jobQueue := filterJobQueue.CreateJobQueue(producer.Output())
-	router := mux.NewRouter()
 
-	s := server.New(cfg.BindAddr, router)
+	apiErrors := make(chan error)
 
-	log.Debug("listening...", log.Data{
-		"bind_address": cfg.BindAddr,
-	})
+	api.CreateFilterAPI(cfg.SecretKey, cfg.Host, cfg.BindAddr, dataStore, &jobQueue, apiErrors)
 
-	_ = api.CreateFilterAPI(cfg.SecretKey, cfg.Host, router, dataStore, &jobQueue)
+	// Gracefully shutdown the application closing any open resources.
+	gracefulShutdown := func() {
+		log.Info(fmt.Sprintf("Shutdown with timeout: %s", cfg.ShutdownTimeout), nil)
+		ctx, cancel := context.WithTimeout(context.Background(), cfg.ShutdownTimeout)
+		defer cancel()
```

Comment:
Not sure if a deferred call to cancel with actually be called because of os.Exit?